lib/BloqadeExpr/src/types.jl

```diff
@@ -310,13 +372,22 @@ function Base.:(==)(lhs::SumOfXPhase, rhs::SumOfXPhase)
     return lhs.nsites == rhs.nsites && lhs.Ω == rhs.Ω && lhs.ϕ == rhs.ϕ
 end
 
+function Base.:(==)(lhs::RydbergHamiltonian, rhs::RydbergHamiltonian)
+    return lhs.rydberg_term == rhs.rydberg_term && lhs.rabi_term == rhs.rabi_term && lhs.detuning_term == rhs.detuning_term
+end
+
 Base.isreal(::RydInteract) = true
 Base.isreal(::SumOfN) = true
 Base.isreal(::SumOfX) = true
 Base.isreal(::SumOfZ) = true
 Base.isreal(::SumOfXPhase) = false
 Base.isreal(h::Add) = all(isreal, subblocks(h))
 Base.isreal(h::Scale) = isreal(factor(h)) && isreal(content(h))
+Base.isreal(h::RydbergHamiltonian{<:Union{Nothing,SumOfX},<:Union{Nothing,SumOfN}}) = true
```

Comment:
you don't have to write the parameters you don't use

```suggestion
Base.isreal(h::RydbergHamiltonian{<:Union{Nothing,SumOfX}}) = true
```